@uportal/action-favorites/src/components/ActionFavorites.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <div :class="'action-favorites ' + fname" @click="toggleFavorite(chanId, $event)"
+       :title="favorite ? translate('message.favorites.remove') : translate('message.favorites.add')">
+    <button class="favorite-button"><icon :name="favorite ? 'star' : 'star-o'" ></icon></button>
+  </div>
+</template>
+
+<script>
+import oidc from "@uportal/open-id-connect";
+import i18n from "../i18n.js";
+import Icon from "vue-awesome/components/Icon";
+import "vue-awesome/icons/star";
+import "vue-awesome/icons/star-o";
+
+const checkStatus = function(response) {
+  if (response.ok) {
+    return response;
+  } else {
+    var error = new Error(response.statusText);
+    error.response = response;
+    throw error;
+  }
+};
+
+const parseJSON = function(response) {
+  return response.json();
+};
+
+export default {
+  name: "ActionFavorites",
+  props: {
+    callOnToggleFav: Function,
+    chanId: { type: String, required: true },
+    favoriteApiUrl: { type: String, default: "/uPortal/api/layout" },
+    fname: String,
+    isFavorite: { type: Boolean, default: false }
+  },
+  data() {
+    return {
+      favorite: this.isFavorite
+    };
+  },
+  components: {
+    Icon
+  },
+  methods: {
+    translate: function(text, lang) {
+      return i18n.t(text, lang);
+    },
+    toggleFavorite: function(chanId, event) {
+      event.preventDefault();
+      if (this.favorite) {
+        this.removeFromFavorite(chanId, event);
+      } else {
+        this.addToFavorite(chanId, event);
+      }
+      return false;
+    },
+    changeFavoriteValue: function() {
+      this.favorite = !this.favorite;
+      this.$emit("is-favorite", this.favorite);
+      if (typeof this.callOnToggleFav === "function") {
+        this.callOnToggleFav(event);
+      }
+    },
+    addToFavorite: function(chanId, event) {
+      oidc().then(token => {
+        const options = {
+          method: "POST",
+          credentials: "same-origin",
+          headers: {
+            Authorization: "Bearer " + token.encoded,
+            "Content-Type": "application/json"
+          }
+        };
+        fetch(
+          this.favoriteApiUrl + "?action=addFavorite&channelId=" + chanId,
+          options
+        )
+          .then(checkStatus)
+          .then(parseJSON)
+          .then(this.changeFavoriteValue(event));
+      });
+      //.catch(err => console.log("Error, with message:", err.statusText));
+    },
+    removeFromFavorite: function(chanId, event) {
```

Comment:
:art: Vue supports `async` functions https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Instructions/async_function
Which can help flatten the promises.